drivers/SmartThings/zigbee-water-leak-sensor/src/test/test_sinope_zigbee_water.lua

```diff
@@ -0,0 +1,314 @@
+-- Copyright 2023 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+-- Mock out globals
+local test = require "integration_test"
+local clusters = require "st.zigbee.zcl.clusters"
+local IASZone = clusters.IASZone
+local TemperatureMeasurement = clusters.TemperatureMeasurement
+local PowerConfiguration = clusters.PowerConfiguration
+local capabilities = require "st.capabilities"
+local zigbee_test_utils = require "integration_test.zigbee_test_utils"
+local IasEnrollResponseCode = require "st.zigbee.generated.zcl_clusters.IASZone.types.EnrollResponseCode"
+local t_utils = require "integration_test.utils"
+
+local ZoneStatusAttribute = IASZone.attributes.ZoneStatus
+
+local mock_device = test.mock_device.build_test_zigbee_device(
+    {
+      profile = t_utils.get_profile_definition("water-temp-battery.yml"),
+      zigbee_endpoints = {
+        [1] = {
+          id = 1,
+          manufacturer = "Sinope Technologies",
+          model = "WL4210",
+          server_clusters = {0x0001, 0x0402, 0x0500}
+        }
+      }
+    }
+)
+zigbee_test_utils.prepare_zigbee_env_info()
+local function test_init()
+  test.mock_device.add_test_device(mock_device)
+  zigbee_test_utils.init_noop_health_check_timer()
+end
+test.set_test_init_function(test_init)
+
+test.register_message_test(
+    "Reported water should be handled: wet",
+    {
+      {
+        channel = "zigbee",
+        direction = "receive",
+        message = { mock_device.id, ZoneStatusAttribute:build_test_attr_report(mock_device, 0x0001) }
+      },
+      {
+        channel = "capability",
+        direction = "send",
+        message = mock_device:generate_test_message("main", capabilities.waterSensor.water.wet())
+      }
+    }
+)
+
+test.register_message_test(
+    "Reported water should be handled: dry",
+    {
+      {
+        channel = "zigbee",
+        direction = "receive",
+        message = { mock_device.id, ZoneStatusAttribute:build_test_attr_report(mock_device, 0x0000) }
+      },
+      {
+        channel = "capability",
+        direction = "send",
+        message = mock_device:generate_test_message("main", capabilities.waterSensor.water.dry())
+      }
+    }
+)
```

Comment:
I'd include a test for the value of `0x0002` sending a "dry" event since that's the primary difference between your driver and the base driver

Reply:
Done